beacon-chain/blockchain/head.go

```diff
@@ -271,9 +271,19 @@ func (s *Service) cacheJustifiedStateBalances(ctx context.Context, justifiedRoot
 	}
 
 	s.clearInitSyncBlocks()
-	justifiedState, err := s.stateGen.StateByRoot(ctx, justifiedRoot)
-	if err != nil {
-		return err
+
+	var justifiedState *stateTrie.BeaconState
+	var err error
+	if justifiedRoot == s.genesisRoot {
+		justifiedState, err = s.beaconDB.GenesisState(ctx)
+		if err != nil {
+			return err
+		}
+	} else {
+		justifiedState, err = s.stateGen.StateByRoot(ctx, justifiedRoot)
+		if err != nil {
+			return err
+		}
 	}
 
 	epoch := helpers.CurrentEpoch(justifiedState)
```

Comment:
Any chance justified state could be nil from the above? Either the db.GenesisState could return nil or stateGen.StateByRoot might. Perhaps needs an extra cehck if this is possible